src/Symfony/Component/Yaml/Yaml.php

```diff
@@ -89,16 +96,23 @@ public static function parse($input, $exceptionOnInvalidType = false, $objectSup
      * @param int   $indent                 The amount of spaces to use for indentation of nested nodes.
      * @param bool  $exceptionOnInvalidType true if an exception must be thrown on invalid types (a PHP resource or object), false otherwise
      * @param bool  $objectSupport          true if object support is enabled, false otherwise
+     * @param bool  $timestampAsDateTime    true if DateTime objects must be dumped as YAML timestamps, false if DateTime objects are not supported
      *
      * @return string A YAML string representing the original PHP array
      *
+     * @deprecated The ability to pass $timestampAsDateTime = false to the Yaml::dump method is deprecated since version 2.8. The argument will be removed in 3.0. Pass true instead.
+     *
      * @api
      */
-    public static function dump($array, $inline = 2, $indent = 4, $exceptionOnInvalidType = false, $objectSupport = false)
+    public static function dump($array, $inline = 2, $indent = 4, $exceptionOnInvalidType = false, $objectSupport = false, $timestampAsDateTime = false)
     {
+        if ($timestampAsDateTime) {
+            trigger_error('The ability to pass $timestampAsDateTime = false to the '.__METHOD__.' method is deprecated since version 2.8. The argument will be removed in 3.0. Pass true instead.', E_USER_DEPRECATED);
```

Comment:
There is nothing about BC here. Your argument does strictly nothing as the code ignores it (except for the deprecation warning being triggered all the time by default, which is bad).
It should simply be removed